### PR TITLE
✂✂✂ Remove Connect

### PR DESCRIPTION
**Pull request by citizen428**
~_This PR will remain in draft until we're ready to remove Connect._~

*we're ready to remove Connect*

## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This PR removes all things related to Connect from the app:

* Routes, corresponding controllers, views, Preact components, etc.
* It drops all Connect related tables
* Completely removes Connect from listings

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [x] This PR changes the Forem platform and our documentation needs to be updated. I have filled out the [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md) issue template so Community Success can help update the Admin Docs appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post (it's possible this is already done by Ben's deprecation notice)
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

If there are any "connect chat is retired" announcements pending on this release they can be handled as this is being released.

PUSHER environment variables can be removed from environments (if we're still in a situation where rollback is conceivable we should keep those, maybe set a task for Friday to revisit this?)

## [optional] What gif best describes this PR or how it makes you feel?

Edward Scissorhands

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -48,6 +48,7 @@ gem "honeycomb-beeline", "~> 2.7.0" # Monitoring and Observability gem
 gem "html_truncator", "~> 0.4" # Truncate an HTML string properly
 gem "htmlentities", "~> 4.3", ">= 4.3.4" # A module for encoding and decoding (X)HTML entities
 gem "httparty", "~> 0.19" # Makes http fun! Also, makes consuming restful web services dead easy
+gem "httpclient", "~> 2.8.3" # Gives something like the functionality of libwww-perl (LWP) in Ruby
```
Comment (citizen428) [inline Gemfile:51]: This was a dependency of Pusher before but when it got removed some thing started failing so I made the dependency explicit now.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -49,8 +49,6 @@
       <body
         data-user-status="<%= user_logged_in_status %>"
         class="<%= Settings::UserExperience.default_font.tr("_", "-") %>-article-body default-header"
-        data-pusher-key="<%= ApplicationConfig["PUSHER_KEY"] %>"
-        data-app-domain="<%= ChatChannel.urlsafe_encoded_app_domain %>"
```
Comment (citizen428) [inline app/views/layouts/application.html.erb:53]: Does anyone know what this was needed for? Is it still required even with Connect gone?

File: config/initializers/reserved_words.rb [hunk 3]
```diff
@@ -33,7 +33,6 @@ class ReservedWords
     challenge
     changelog
     chat
```
Comment (citizen428) [inline config/initializers/reserved_words.rb:35]: Should "chat" remain a reseved word?

**Comment by benhalpern** [comment]
This is looking good from my perspective. Let's make sure to give our community success team ample time to move at their pace, and we can finish this up and be ready to move on their mark. They're working on being ready for this, but let's not accidentally move forward without conversing and making sure the rest of the communication is in place.

**Comment by citizen428** [comment]
@benhalpern Oh for sure, I just want us to be ready for the removal. There are still some unsolved product questions too which I'll pop in a Slack channel.

**Comment by maestromac** [comment]
I believe the content within `app/views/mailers/notify_mailer/channel_invite_email.text.erb` should be also removed?

**Comment by djuber** [comment]
Reviewers: 
- chat/connect functionality should not be present in app (mentions or links to chat are probably defects, feel free to point this out)
- main merge was larger than I had hoped - if there's a change in meaning between https://github.com/forem/forem/pull/14734/commits/5f5dfaed9a5a8cbe0a817bce71411476ae87d0f9 and HEAD that's probably my fault!
- while the build is green - I had some problems with pre-commit hooks/linters in the main merge PR and used `--no-verify` to skip them after fighting with it. The CI build is green but I'm unsure that's a good measure of quality.

File: Gemfile (hunk 1 above)
Comment (djuber) [reply Gemfile]: missed this in the main merge - added this back (probably without the comment, will add)

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (djuber) [reply app/views/layouts/application.html.erb:54]: I can't find any reference to this app domain attribute anywhere, and the encoded app domain is only used here and in pusher_channels.

**Comment by djuber** [comment]
Question:

We had a route /connect which loaded the functionality.

Should we add a _page_ with "connect" as a slug to handle the retirement? Right now, as soon as this is merged, you'll get a 404 page. Seems like adding a redirect from /connect (as a page slug) to Ben's post about removing this feature makes sense.

File: config/initializers/reserved_words.rb (hunk 3 above)
Comment (jeremyf) [reply config/initializers/reserved_words.rb:35]: I think for now…it's alright.

**Comment by djuber** [comment]
documentation status: 

admin docs

https://github.com/forem/admin-docs/issues/22 was closed/done - the linked [documentation page](https://admin.forem.com/docs/advanced-customization/chat-channels) for chat channels gives a 404 (correct) and the connect reference was removed from listings in https://github.com/forem/admin-docs/commit/14a10e77fb56b37e2a8cc80d3cf7268065c50069

**Review by jeremyf — APPROVED** [review]
That's a lot of removed code.  By my understanding, if things go wonky, the patches wouldn't be hard to roll out.

**Review by jeremyf — APPROVED** [review]

**Review by djuber — APPROVED** [review]
Approving, but I had the last set of commits today and have self-assigned as well, should include at least 2 other reviewers before merging.

**Comment by djuber** [comment]
If there's a deadline to disable this in DEV.to, that should be handled with the "connect" feature flag, this can release when it's ready (maybe today, maybe tomorrow morning) independently of turning off the functionality.

**Review by nickytonline — APPROVED** [review]
Took it for a spin, nothing breaks and nothing stands out in the browser console. ✂️  🚀

**Comment by djuber** [comment]
planning to merge this in the morning - feature flag for connect was disabled in DEV.to at 22:39 UTC (20 minutes ago).

**Comment by jeremyf** [comment]
Thank you @djuber on stewarding this.

**Pull request merged**